### ISSUE TITLE
MaxPool layer does not correctly implement padding.

### ISSUE THREAD

**Issue by nmcclatchey** [pytorch/pytorch#41294]
## 🐛 Bug

Per [documentation of MaxPool](https://pytorch.org/docs/stable/nn.html#maxpool2d), input is "implicitly zero-padded on both sides for padding number of points." The implementation differs by implicitly padding with identity elements, rather than zero.

## To Reproduce

1. Create a tensor of signed type, filled with negative values.
1. Invoke a MaxPool with positive padding.
1. Compare against ground truth, either computed by hand or created by explicit zero-padding.

Minimal code sample / unit test:
```
# Maxpool of non-positive values, such that padding is included in the computation of each result element.
import torch
r = torch.nn.functional.max_pool2d(-torch.rand(2,2,64), kernel_size=[3,3], padding=1)
assert torch.equal(r, torch.zeros_like(r))
```

## Expected behavior

According to documentation, the values near the edge will be computed using the implicit zero-padding. The maximum of a negative number and zero is zero, so such values _ought_ to be zero. In the provided example, the assert should not fire.

When the code sample above is run, however, the assert fires.

## Environment

Information from collect_env.py:

- PyTorch version: 1.5.1
- Is debug build: No
- CUDA used to build PyTorch: 10.2

- OS: Microsoft Windows 10 Home
- GCC version: (x86_64-win32-seh-rev0, Built by MinGW-W64 project) 8.1.0
- CMake version: version 3.14.1

- Python version: 3.8
- Is CUDA available: Yes
- CUDA runtime version: Could not collect
- GPU models and configuration: GPU 0: GeForce GTX 960M
- Nvidia driver version: 442.50
- cuDNN version: Could not collect

- Versions of relevant libraries:
- [pip] Could not collect
- [conda] Could not collect

## Additional context

N/A

cc @ezyang @gchanan @zou3519 @jlin27 @albanD @mruberry

**Comment by albanD** [comment]
This is a duplicate of https://github.com/pytorch/pytorch/issues/33384. 
I'm going to close this to keep all discussions there.
